session.go

```diff
@@ -390,7 +397,7 @@ func (s *Session) waitForSendErr(hdr header, body io.Reader, errCh chan error, t
 
 WAIT:
 	select {
-	case <-s.shutdownCh:
+	case <-s.sendDoneCh: // shutdown isn't enough, we need to wait for the send loop to exit.
```

Comment:
Got it. This prevents `waitForSendErr` from returning too early and allowing a new call to slink in while the send loop is still running 👍